third_party/parser/cc/lexer.rl

```diff
@@ -1181,6 +1181,25 @@ void lexer::set_state_expr_value() {
     fcall expr_variable;
   }
 
+  # Special case for Ruby > 2.7
+  # If interpolated instance/class variable starts with a digit we parse it as a plain substring
+  # However, "#$1" is still a regular interpolation
+  interp_digit_var = '#' ('@' | '@@') digit c_alpha*;
+
+  action extend_interp_digit_var {
+    if (version >= ruby_version::RUBY_27) {
+      auto& current_literal = literal_();
+      std::string str = tok();
+      current_literal.extend_string(str, ts, te);
+    } else {
+      if (ts[0] == '#' && ts[1] == '@' && ts[2] == '@') {
```

Comment:
We're maintaining a lot of code that is actually never executed. Are we planing to allow Sorbet to handle multiple versions?

Reply:
I removed that commit